state/executor.go

```diff
@@ -182,6 +182,13 @@ func (e *BlockExecutor) Commit(ctx context.Context, state types.State, block *ty
 	if err != nil {
 		e.logger.Error("failed to fire block events", "error", err)
 	}
+	// this channel is passed in by lazy aggregators,
+	// to notify it that transactions remain in the mempool after building a block.
+	if notifyMempoolNotEmpty != nil {
+		if e.mempool.Size() > 0 {
+			notifyMempoolNotEmpty <- struct{}{}
+		}
+	}
```

Comment:
[question][non-blocking] Can you explain why the lazy sequencer needs to be notified of the transactions remaining in the mempool after building a block?

Reply:
The idea was if the number of transactions that arrive in 1 second is larger than the blocksize, then keep building blocks until mempool is empty.

@gupadhyaya pointed out in another comment that this may not be worth the added complexity, so I will remove this

Reply:
Removed this feature to reduce complexity- will write in a follow-up issue and we can do something if it turns out to be a problem.

Reply:
https://github.com/rollkit/rollkit/issues/777